### PR TITLE
K8SPSMDB-338 PMM2.0 for openshift

### PR DESCRIPTION
**Pull request by cap1984**
[K8SPSMDB-338](https://jira.percona.com/browse/K8SPSMDB-338)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | failed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| operator-self-healing | passed |
| security-context | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| arbiter | passed |
| service-per-pod | failed |
| liveness | failed |

File: e2e-tests/monitoring/run [hunk 1]
```diff
@@ -13,7 +13,7 @@ deploy_operator
```
Comment (hors) [inline e2e-tests/monitoring/run:13]: we need to remove this test at all

File: e2e-tests/monitoring-2-0/run [hunk 2]
```diff
@@ -13,13 +13,19 @@ deploy_operator
...
-    platform=openshift
-fi
+
 helm del --purge monitoring || :
 helm repo remove stable || :
 helm repo add stable https://charts.helm.sh/stable
-retry 10 60 helm install monitoring --set platform=$platform https://percona-charts.storage.googleapis.com/pmm-server-2.12.0.tgz
+if [[ ! -z "${OPENSHIFT}" ]]; then
+    platform=openshift
+    oc adm policy add-scc-to-user privileged -z percona-server-mongodb-operator
```
Comment (hors) [inline e2e-tests/monitoring-2-0/run:22]: I think we need to rollback these changes at the end of test.

**Review by hors — APPROVED** [review]

**Review by hors — CHANGES_REQUESTED** [review]
please check comments

File: e2e-tests/functions [hunk 3]
```diff
@@ -20,13 +20,17 @@ conf_dir=$(realpath $test_dir/../conf || :)
...
+        "v1.11") OPENSHIFT=3.11 ;;
+              *) OPENSHIFT=4    ;;
+    esac
 fi
 
 
 create_namespace() {
     local namespace="$1"
-    if [ "$OPENSHIFT" == 1 ]; then
+    if [ ! -z "${OPENSHIFT}" ]; then
```
[the remark this answers is not in the document]
Comment (cap1984) [reply e2e-tests/functions]: done

[the remark this answers is not in the document]
Comment (cap1984) [reply e2e-tests/functions]: done

File: e2e-tests/functions [hunk 4]
```diff
@@ -331,8 +335,11 @@ compare_kubectl() {
     local expected_result=${test_dir}/compare/${resource//\//_}${postfix}.yml
     local new_result="${tmp_dir}/${resource//\//_}.yml"
 
-    if [ "$OPENSHIFT" = 1 -a -f ${expected_result//.yml/-oc.yml} ]; then
+    if [ ! -z "$OPENSHIFT" -a -f ${expected_result//.yml/-oc.yml} ]; then
```
[the remark this answers is not in the document]
Comment (cap1984) [reply e2e-tests/functions]: done

File: e2e-tests/functions [hunk 5]
```diff
@@ -532,7 +539,7 @@ destroy() {
     kubectl_bin delete psmdb-restore --all || :
 
     kubectl_bin delete -f https://github.com/jetstack/cert-manager/releases/download/v0.15.1/cert-manager.yaml 2>/dev/null || :
-    if [ "$OPENSHIFT" == 1 ]; then
+    if [ ! -z "$OPENSHIFT" ]; then
```
[the remark this answers is not in the document]
Comment (cap1984) [reply e2e-tests/functions]: done

File: e2e-tests/security-context/run [hunk 6]
```diff
@@ -32,7 +32,7 @@ wait_for_running "${cluster}-rs0" "3"
 compare_kubectl "statefulset/${cluster}-rs0" $postfix
 compare_kubectl "cronjob.batch/${cluster}-backup-each-hour"
 
-if [[ "${OPENSHIFT}" == 1 ]]; then
+if [[ ! -z "${OPENSHIFT}" ]]; then
```
[the remark this answers is not in the document]
Comment (cap1984) [reply e2e-tests/security-context/run]: done

File: e2e-tests/security-context/run [hunk 7]
```diff
@@ -11,7 +11,7 @@ deploy_operator
 
 kubectl_bin apply -f $conf_dir/secrets.yml -f $conf_dir/client.yml -f $conf_dir/minio-secret.yml
 kubectl_bin apply -f "$test_dir/conf/service-account.yml"
-if [[ "${OPENSHIFT}" == 1 ]]; then
+if [[ ! -z "${OPENSHIFT}" ]]; then
```
[the remark this answers is not in the document]
Comment (cap1984) [reply e2e-tests/security-context/run]: done

File: e2e-tests/monitoring/run [hunk 8]
```diff
@@ -13,7 +13,7 @@ deploy_operator
 
 desc 'install PMM Server'
 platform=kubernetes
-if [ "$OPENSHIFT" == 1 ]; then
+if [ ! -z "$OPENSHIFT" ]; then
```
[the remark this answers is not in the document]
Comment (cap1984) [reply e2e-tests/monitoring/run]: done

File: e2e-tests/monitoring-2-0/run [hunk 9]
```diff
@@ -13,13 +13,19 @@ deploy_operator
...
 platform=kubernetes
-if [ "$OPENSHIFT" == 1 ]; then
-    platform=openshift
-fi
+
 helm del --purge monitoring || :
 helm repo remove stable || :
 helm repo add stable https://charts.helm.sh/stable
-retry 10 60 helm install monitoring --set platform=$platform https://percona-charts.storage.googleapis.com/pmm-server-2.12.0.tgz
+if [[ ! -z "${OPENSHIFT}" ]]; then
```
[the remark this answers is not in the document]
Comment (cap1984) [reply e2e-tests/monitoring-2-0/run]: done

File: e2e-tests/monitoring-2-0/run (hunk 2 above)
Comment (cap1984) [reply e2e-tests/monitoring-2-0/run:22]: done

File: e2e-tests/monitoring/run (hunk 1 above)
Comment (cap1984) [reply e2e-tests/monitoring/run]: done

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-505-27bf3c56

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| one-pod | passed |
| demand-backup | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| service-per-pod | passed |
| liveness | passed |
| demand-backup-sharded | passed |
| smart-update | passed |
| upgrade | passed |
| version-service | passed |
| upgrade-sharded | passed |
| users | passed |

**Review by tplavcic — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Pull request merged**